composer/loggers/tqdm_logger.py

```diff
@@ -101,73 +102,37 @@ def _will_log(self, state: State, log_level: LogLevel) -> bool:
-                                              keys_to_log=["loss/train"])
-
-    def after_backward(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_train is not None
-        self.pbar_train.update()
-
-    def epoch_end(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_train is not None
-        self.pbar_train.close()
-        self.pbar_train = None
-
-    def eval_start(self, state: State, logger: Logger) -> None:
-        del logger  # Unused
-        assert self.pbar_val is None
-        assert state.eval_dataloader is not None
-        self.pbar_val = _TQDMLoggerInstance(total=len(state.eval_dataloader),
-                                            epoch=state.epoch,
-                                            val=True,
-                                            keys_to_log=["accuracy/val"])
-        self.is_validating = True
-
-    def eval_after_forward(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_val is not None
-        self.pbar_val.update()
-
-    def eval_end(self, state: State, logger: Logger) -> None:
-        del state, logger  # Unused
-        assert self.pbar_val is not None
-        self.pbar_val.close()
-        self.pbar_val = None
-        self.is_validating = False
+        self.active_pbar.log_metric(data)
+
+    def _run_event(self, event: Event, state: State, logger: Logger) -> None:
+        if event == Event.TRAINING_START:
+            if self.config is not None:
+                print("Config")
+                print("-" * 30)
+                yaml.safe_dump(self.config, stream=sys.stdout)
+                print("-" * 30)
+                print()
+        if event in (Event.EPOCH_START, Event.EVAL_START):
+            is_train = event == Event.TRAINING_START
+            self.pbars[is_train] = _TQDMLoggerInstance(total=state.steps_per_epoch,
+                                                       epoch=state.epoch,
+                                                       is_train=is_train)
+            self.active_pbar = self.pbars[is_train]
+        if event in (Event.AFTER_BACKWARD, Event.EVAL_AFTER_FORWARD):
+            assert self.active_pbar is not None
+            self.active_pbar.update()
+        if event in (Event.EPOCH_END, Event.EVAL_END):
+            assert self.active_pbar is not None
+            self.active_pbar.close()
+            self.active_pbar = None
+
+        super()._run_event(event, state, logger)
 
     def state_dict(self) -> StateDict:
-        state = {"is_validating": self.is_validating}
-        if self.pbar_train:
-            state["pbar_train"] = self.pbar_train.state_dict()
-        if self.pbar_val:
-            state["pbar_val"] = self.pbar_val.state_dict()
-        return state
+        return {"pbars": {k: v.state_dict() for (k, v) in self.pbars.items()}}
```

Comment:
Does the `state_dict` not need to store `active_pbar`? I guess it probably wasn't possible to checkpoint in the middle of validation, but it's totally possible to checkpoint in the middle of a batch and I'm wondering if that can lead to subtle issues when restoring from a checkpoint.

Reply:
(Yup, added it back in)